spec/fixtures/custom_nginx.template

```diff
@@ -245,6 +245,14 @@ http {
 
         set_real_ip_from 127.0.0.1;
 
+        location / {
+            content_by_lua_block {
+                local mu = require "spec.fixtures.mock_upstream"
+                ngx.status = 404
+                return mu.send_default_json_response()
+            }
+        }
+
```

Comment:
Is this extra block necessary? can't it be solved with a dynamic mock instead of growing the test config file with yet another entry? See recently merged https://github.com/Kong/kong/pull/4454

Reply:
I think having usable `404` endpoint without any injections is a good addition. The `404` endpoint previously just returned some default `OpenResty` html. This makes `404` a way more usable.

Reply:
ah ok, so this is generic and not tight to this PR specifically.